activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -163,7 +163,18 @@ def migrations_paths # :nodoc:
       end
 
       def migration_context # :nodoc:
-        MigrationContext.new(migrations_paths)
+        MigrationContext.new(migrations_paths, schema_migration)
+      end
+
+      def schema_migration
+        @schema_migration ||= begin
+                                conn = self
+                                Class.new(ActiveRecord::SchemaMigration) do
+                                  self.table_name = "schema_migrations"
```

Comment:
Is this necessary? Actually the `SchemaMigration` super class is not abstract class.

Reply:
Removed